plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java

```diff
@@ -234,6 +259,127 @@ public void onFailure(Exception e) {
+                org.opensearch.ingest.PipelineConfiguration::getId
+            );
+            if (allDependentPipelineIds.isEmpty()) {
+                actionListener.onResponse(true);
+            } else {
+                actionListener
+                    .onFailure(
+                        new OpenSearchStatusException(
+                            String
+                                .format(
+                                    Locale.ROOT,
+                                    "%d ingest pipelines are still using this model, please delete or update the pipelines first: %s",
+                                    allDependentPipelineIds.size(),
+                                    Arrays.toString(allDependentPipelineIds.toArray(new String[0]))
+                                ),
+                            RestStatus.CONFLICT
+                        )
+                    );
+            }
+        }, e -> {
+            log.error("Failed to delete ML Model: " + modelId, e);
+            actionListener.onFailure(e);
+
+        }));
+
+    }
+
+    private void checkSearchPipelineBeforeDeleteModel(String modelId, ActionListener<Boolean> actionListener) {
+        GetSearchPipelineRequest getSearchPipelineRequest = new GetSearchPipelineRequest();
+        client.execute(GetSearchPipelineAction.INSTANCE, getSearchPipelineRequest, ActionListener.wrap(searchPipelineResponse -> {
+            List<String> allDependentPipelineIds = findDependentPipelines(
+                searchPipelineResponse.pipelines(),
+                modelId,
+                org.opensearch.search.pipeline.PipelineConfiguration::getConfigAsMap,
+                org.opensearch.search.pipeline.PipelineConfiguration::getId
+            );
+            if (allDependentPipelineIds.isEmpty()) {
+                actionListener.onResponse(true);
+            } else {
+                actionListener
+                    .onFailure(
+                        new OpenSearchStatusException(
+                            String
+                                .format(
+                                    Locale.ROOT,
+                                    "%d search pipelines are still using this model, please delete or update the pipelines first: %s",
+                                    allDependentPipelineIds.size(),
+                                    Arrays.toString(allDependentPipelineIds.toArray(new String[0]))
+                                ),
+                            RestStatus.CONFLICT
+                        )
+                    );
+            }
+        }, e -> {
+            log.error("Failed to delete ML Model: " + modelId, e);
+            actionListener.onFailure(e);
+
+        }));
+
+    }
+
+    private void checkDownstreamTaskBeforeDeleteModel(String modelId, Boolean isHidden, ActionListener<DeleteResponse> actionListener) {
+        // Now 3 is: 1. Agent 2. Search pipeline 3. ingest pipeline
```

Comment:
How about something along the lines of checks 3 resources associated with with the model id 1. Agent 2. Search pipeline 3. ingest pipeline

Reply:
Already changed.